Geometry/MTDGeometryBuilder/plugins/MTDParametersESModule.cc

```diff
@@ -25,25 +26,40 @@ class MTDParametersESModule : public edm::ESProducer {
 
 private:
   MTDParametersFromDD builder;
-  const edm::ESGetToken<DDCompactView, IdealGeometryRecord> compactViewToken_;
+
+  const bool fromDD4hep_;
+  edm::ESGetToken<DDCompactView, IdealGeometryRecord> ddCompactToken_;
+  edm::ESGetToken<cms::DDCompactView, IdealGeometryRecord> dd4hepToken_;
 };
 
 MTDParametersESModule::MTDParametersESModule(const edm::ParameterSet& pset)
-    : compactViewToken_{setWhatProduced(this).consumesFrom<DDCompactView, IdealGeometryRecord>(edm::ESInputTag())} {
-  edm::LogInfo("TRACKER") << "MTDParametersESModule::MTDParametersESModule";
+    : fromDD4hep_(pset.getParameter<bool>("fromDD4hep")) {
+  auto cc = setWhatProduced(this);
+  if (!fromDD4hep_) {
+    ddCompactToken_ = cc.consumesFrom<DDCompactView, IdealGeometryRecord>(edm::ESInputTag());
+  } else {
+    dd4hepToken_ = cc.consumesFrom<cms::DDCompactView, IdealGeometryRecord>(edm::ESInputTag());
+  }
+  edm::LogInfo("MTDParametersESModule") << "MTDParametersESModule::MTDParametersESModule";
```

Comment:
Just noticing this line. Shouldn't it be `LogTrace`?

Reply:
Throughout this PR, shouldn't the LogInfo/LogVerbatim be changed to `LogTrace`?

This PR has already been merged, so these changes would have to go in a later PR.

Reply:
@cvuosalo this is a minimal update of older code, I agree with you that it might be set as optional (even though it is not so far). Anyway 1) Info verbosity is not used in production 2) LogDebug/LogTrace management looks to me quite impractical and cumbersome and 3) it is basically not tested (I have recently made a cleanup PR for many LogDebug  lines not even compiling when invoked, because they are generally not probed in regular PR tests). I can move there printouts under  EDM_ML_DEBUG flag control as usual in #29647

Reply:
> it is basically not tested (I have recently made a cleanup PR for many LogDebug lines not even compiling when invoked, because they are generally not probed in regular PR tests). I can move there printouts under EDM_ML_DEBUG flag control as usual in #29647h

Code under `EDM_ML_DEBUG` has the same risks towards bitrot as `LogDebug`, so the only benefit of not using something else than `LogDebug` inside `#ifdef EDM_ML_DEBUG` is possibly a bit easier configuration of the MessageLogger (`LogInfo` needs some as well).

Reply:
@makortel in this respect you are correct, the simpler update of the default MessageLogger configuration to switch on/off messages is the main reason for this choice of mine (in general we want to comment the declarations of EDM_ML_DEBUG in the code, so my comment 3 is not really relevant in that case). In #29250 my suggestion was to implement a check of the output of compilation with that flag, enabled for the static analyzer, which I understand has been implemented by @smuzaffar in https://github.com/cms-sw/cms-bot/pull/1289